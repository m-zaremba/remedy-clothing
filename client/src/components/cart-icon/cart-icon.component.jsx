import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';
import { CartIconContainer, ShopIcon, ItemCount } from './cart-icon.styles';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

export const CartIcon = ({ toggleCartHidden, itemCount }) => {
  return (
    <CartIconContainer
      role="button"
      onClick={toggleCartHidden}
      onKeyDown={toggleCartHidden}
      tabIndex={0}
    >
      <ShopIcon />
      <ItemCount>{itemCount}</ItemCount>
    </CartIconContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

CartIcon.propTypes = {
  toggleCartHidden: PropTypes.func.isRequired,
  itemCount: PropTypes.number.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
