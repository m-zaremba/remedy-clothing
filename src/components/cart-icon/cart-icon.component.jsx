import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';
import './cart-icon.styles.scss';
import { ReactComponent as ShoppingIcon } from '../../assets/shopping-bag.svg';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

const CartIcon = ({ toggleCartHidden, itemCount }) => {
  return (
    <div
      role="button"
      className="cart-icon"
      onClick={toggleCartHidden}
      onKeyDown={toggleCartHidden}
      tabIndex={0}
    >
      <ShoppingIcon className="shopping-icon" />
      <span className="item-count">{itemCount}</span>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

CartIcon.propTypes = {
  toggleCartHidden: PropTypes.func.isRequired,
  itemCount: PropTypes.number.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
