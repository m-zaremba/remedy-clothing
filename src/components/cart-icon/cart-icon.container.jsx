import React from 'react';
import { graphql } from 'react-apollo';
import { gql } from 'apollo-boost';
import { flowRight } from 'lodash';

import CartIcon from './cart-icon.component';

const TOGGLE_CART_HIDDEN = gql`
  mutation ToggleCartHidden {
    toggleCartHidden @client
  }
`;

const GET_CART_ITEMS_COUNT = gql`
  {
    cartItemsCount @client
  }
`;

const CartIconContainer = ({ data: { cartItemsCount }, toggleCartHidden }) => (
  <CartIcon toggleCartHidden={toggleCartHidden} cartItemsCount={cartItemsCount} />
);

export default flowRight(
  graphql(GET_CART_ITEMS_COUNT),
  graphql(TOGGLE_CART_HIDDEN, { name: 'toggleCartHidden' }),
)(CartIconContainer);
