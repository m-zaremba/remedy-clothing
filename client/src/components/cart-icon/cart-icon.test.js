import React from 'react';
import { shallow } from 'enzyme';
import { CartIcon } from './cart-icon.component';

describe('testing CartIcon component', () => {
  let wrapper;
  let mockToggleCartHidden;
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden,
    };

    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it('should render CartIcon component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartIconContainer').simulate('click');
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it('should render the itemCount value', () => {
    const itemCount = parseInt(wrapper.find('ItemCount').text(), 10);
    expect(itemCount).toBe(0);
  });
});
