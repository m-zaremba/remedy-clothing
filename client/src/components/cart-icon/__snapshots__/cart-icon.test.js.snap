// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CartIcon component should render CartIcon component 1`] = `
<CartIconContainer
  onClick={[MockFunction]}
  onKeyDown={[MockFunction]}
  role="button"
  tabIndex={0}
>
  <Styled(Component) />
  <ItemCount>
    0
  </ItemCount>
</CartIconContainer>
`;
